src/app/app.component.html

```diff
@@ -1 +1,108 @@
+<main class="main">
+    <form class="register-form" [formGroup]="registerForm" (ngSubmit)="onSubmit()">
+        <p class="title">Reactive Forms</p>
+        <mat-form-field class="example-full-width" appearance="fill">
+            <mat-label>Name</mat-label>
+            <input matInput placeholder="Enter name" formControlName="firstName">
+            <mat-error *ngIf="registerForm.controls.firstName.hasError('required')">
+                Name is <strong>required</strong>
+            </mat-error>
+            <mat-error *ngIf="registerForm.controls.firstName.hasError('minlength')">
+                The value you inputted is 
+                {{ registerForm.controls.firstName.errors?.minlength.actualLength }} 
+                characters long.
+                It must be at least 
+                {{ registerForm.controls.firstName.errors?.minlength.requiredLength }} 
+                characters long.
+            </mat-error>
+            <mat-error *ngIf="registerForm.controls.firstName.hasError('maxlength')">
+                The value you passed is too long.
+            </mat-error>
+        </mat-form-field>
 
+        <mat-form-field class="example-full-width" appearance="fill">
+            <mat-label>Last Name</mat-label>
+            <input matInput placeholder="Enter Last Name" formControlName="lastName">
+            <mat-error *ngIf="registerForm.controls.lastName.hasError('required')">
```

Comment:
you usually have check on required error, let's create function with field param that checks it

Reply:
I have some questions(doubts) and ideas about it. 
So I didn`t implement this in final version of app.